<!-- 
  Simulate a differential drive robot with the teb_local_planner in stage:
  - stage
  - map_server
  - move_base
  - static map
  - amcl
  - rviz view
 -->
<launch>
  <remap from="odom" to="/odometry/filtered"/>
  <remap from="/cmd_vel" to="roboteq_driver/cmd"/>
  <arg name="cutting_spacing" default="0.4" />
  <arg name="map_file" default="$(find robot_mower_2dnav)/cfg/map.yaml"/>
        <!--  ************** Navigation ***************  -->
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
  	  	<rosparam file="$(find robot_mower_2dnav)/cfg/costmap_common_params.yaml" command="load" ns="global_costmap" />
  	 	<rosparam file="$(find robot_mower_2dnav)/cfg/costmap_common_params.yaml" command="load" ns="local_costmap" />
  		<rosparam file="$(find robot_mower_2dnav)/cfg/local_costmap_params.yaml" command="load" />
  		<rosparam file="$(find robot_mower_2dnav)/cfg/global_costmap_params.yaml" command="load" />
  		<rosparam file="$(find robot_mower_2dnav)/cfg/teb_local_planner_params.yaml" command="load" />

		<param name="base_global_planner" value="carrot_planner/CarrotPlanner" />
		<param name="allow_unknown" value="true" />
		<param name="planner_frequency" value="3.0" />
		<param name="planner_patience" value="5.0" />

		<param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
		<param name="controller_frequency" value="5.0" />
		<param name="controller_patience" value="15.0" />
	</node>
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />
    <node name="heatmap" pkg="robot_mower_2dnav" type="heatmap_client" respawn="false" output="screen"/>
    <node name="waypoint" pkg="robot_mower_2dnav" type="path_planner.py" respawn="false" output="screen"/>
</launch>
